zhaquirks/xiaomi/aqara/switch_acn047.py

```diff
@@ -54,14 +54,15 @@ def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
 
     def _update_attribute(self, attrid, value):
-        if attrid == STATUS_TYPE_ATTR:
-            self._current_state = PRESS_TYPES.get(value)
+        if attrid == STATUS_TYPE_ATTR and value == 1:
```

Comment:
Just to make sure I understand, what is `value` for this device when the presses are triggered? And why do you check for `1`?

Reply:
Anyhow I shorted the pins (short, long, ultralong, double, triple etc) I always received 1. So I did not want to check value for 1 but the original implementation did check in a way only index 1 was available in the PRESS_TYPES hash/object. So to be safe 1 checked.

What I could do is to create a nested hash, so the button event names are accessible this way: PRESS_TYPES[endpoint].get(value) so the check for fix 1 could be removed. Wdyt?

Reply:
Hmm, okay. I'd leave it like this for now then. Should be fine, as we don't seem to ignore any events.
(But it would be good to find out if it can ever be something other than 1)

Reply:
I am not sure what are the best practices in this repo but what we could do is to surface "unknown" for the users so they can notice and report it. I guess they can report also if something is not working correctly. I am not a big fan of hardcoding stuff, we could also expose the raw events so it is arbitrary, for example: `event_1_1`, `event_1_2` in the current scenario. Or we can do it when `switch_1` and `switch_2` is not recognised. 🙃 

If we are here, could you please pass me a documentation/reference PR so I can learn how to extend HA to include `interlock` and `pulse_length` inputs on the device UI? 😊

Reply:
> we could do is to surface "unknown" [...] also expose the raw events

I think we generally don't do this. I'd still leave it like this for now (or remove the `value == 1` part?).
Is there a difference when it's in dry mode or not? 

Maybe the answer is somewhere in the zigbee-herdsman-converters library, but I wasn't able to find it quickly.

---

For HA entities:

So this is already implemented for the T2 relay: [homeassistant/components/zha/select.py#L426-L435](https://github.com/home-assistant/core/blob/5dac5d5c7e9c5a1bf3ec4edc5198a7a0d92d27ee/homeassistant/components/zha/select.py#L426-L435)

For interlock, you just want on/off, so a switch in HA. Here and something like this: [homeassistant/components/zha/switch.py#L293-L302](https://github.com/home-assistant/core/blob/5dac5d5c7e9c5a1bf3ec4edc5198a7a0d92d27ee/homeassistant/components/zha/switch.py#L293-L302), but with the model and attribute changed, maybe also icon.

You also need to add the translation key to the JSON strings file and generate the translations locally: https://developers.home-assistant.io/docs/internationalization/core/#test-translations

For pulse length, look at `number.py` (custom entities are all towards the bottom of each file.

This will be easier to expose _eventually_ (just with quirks).

Reply:
> Is there a difference when it's in dry decoupled mode or not?

Yes, this events will fire only if it is in decoupled mode.